### PR TITLE
removed serve v1 support

### PR DESCRIPTION
**Pull request by blublinsky**
## Why are these changes needed?

As operator deprecates serve V1 support, so does an API server
## Related issue number
fixes https://github.com/ray-project/kuberay/issues/1829

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ x] Unit tests
   - [ x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by blublinsky** [comment]
This is WIP

**Comment by blublinsky** [comment]
@z103cb Please take a look

cc: @kevin85421

File: apiserver/test/e2e/job_submission_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
+package e2e
+
+import (
+	"testing"
+
+	api "github.com/ray-project/kuberay/proto/go_client"
+	"github.com/stretchr/testify/require"
+)
+
+// TestCreateClusterAutoscalerEndpoint sequentially iterates over the create cluster endpoint
```
Comment (z103cb) [inline apiserver/test/e2e/job_submission_e2e_test.go:10]: I don't think the comment makes sense.

**Review by z103cb — COMMENTED** [review]
Small nitpick on a comment. Otherwise it looks good.

**Review by z103cb — COMMENTED** [review]
LGTM.

**Comment by blublinsky** [comment]
@kevin85421 all yours to commit

**Review by z103cb — APPROVED** [review]
LGTM

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**